scripts/renderers.py

```diff
@@ -43,7 +43,11 @@ def add_info_source_to_dict_keys(self, toggle_info_structured_dicts):
         """
         temp_data = []
         for datum in toggle_info_structured_dicts:
-            toggle_dict_info = {}
+            base_info_keys = list(datum.keys())
+            # removing state and annotations keys cause those will be taken care of below
+            base_info_keys.remove("state")
+            base_info_keys.remove("annotations")
+            toggle_dict_info = {key:value for key, value in datum.items() if key in base_info_keys}
```

Comment:
Suggestion: If you instead do `toggle_dict_info = datum.copy()` you can then do `toggle_dict_info.pop('state', None)` and `toggle_dict_info.pop('annotations', None)` after the for-loops. It might communicate intent a bit more clearly.

Optional, though.

Reply:
(Using `pop` this way is a little naughty, but Python lacks a concise, straightforward key-removal method for dictionaries...)